Objects/Converters/ConverterAutocadCivil/ConverterAutocadCivilShared/ConverterAutocadCivil.Civil.cs

```diff
@@ -736,44 +736,48 @@ private Featureline FeaturelineToSpeckle(CivilDB.CorridorFeatureLine featureline
   public Mesh SurfaceToSpeckle(TinSurface surface)
   {
     // output vars
-    List<Point3d> vertices = new();
-    var faces = new List<int>();
+    List<double> vertices = new();
+    List<int> faces = new ();
+    Dictionary<Point3d, int> indices = new();
+    
+    int indexCounter = 0;
     foreach (var triangle in surface.GetTriangles(false))
     {
-      var triangleVertices = new List<Point3d> { triangle.Vertex1.Location, triangle.Vertex2.Location, triangle.Vertex3.Location };
-
-#if CIVIL2023 || CIVIL2024 // skip any triangles that are hidden in the surface!
-      if (!triangle.IsVisible)
-      {
-        triangle.Dispose();
-        continue;
-      }
-#endif
-
-      // store vertices
-      var faceIndices = new List<int>();
-      foreach (var vertex in triangleVertices)
+      try
       {
-        if (!vertices.Contains(vertex))
+#if CIVIL2023 || CIVIL2024 // skip any triangles that are hidden in the surface!
+        if (!triangle.IsVisible)
         {
-          faceIndices.Add(vertices.Count);
-          vertices.Add(vertex);
+          //TODO: why are we skipping trianlges like this, are we not excluding them already in our call to GetTriangles(false) ?
```

Comment:
True - no need to have this codeblock then!

Reply:
Removed!